t/lib/MakeMaker/Test/NoXS.pm

```diff
@@ -13,12 +13,13 @@ require XSLoader;
 delete $DynaLoader::{boot_DynaLoader};
 
 if ($^O eq 'MSWin32') {
+    # Load Win32. Backup GetCwd and SetChildShowWindow. Then clear the stash of all other entries.
     require Win32;
-    my $GetCwd = *{'Win32::GetCwd'}{CODE};
-    my $SetChildShowWindow = *{'Win32::SetChildShowWindow'}{CODE};
-    %{*main::Win32::{HASH}} = ();
-    *{'Win32::GetCwd'} = $GetCwd;
-    *{'Win32::SetChildShowWindow'} = $SetChildShowWindow;
+    my $GetCwd = Win32->can('GetCwd');
+    my $SetChildShowWindow = Win32->can('SetChildShowWindow');
+    %Win32:: = ();
+    *Win32::GetCwd = $GetCwd;
```

Comment:
This doesn't work. It binds early, so the glob it's assigning to gets erased by the stash being cleared.

Reply:
`*Win32::GetCwd` binds to the glob at compile time. Clearing the stash removes that glob from the namespace, but because it's already bound, the assignment stores the sub in a glob that isn't in the namespace anymore.

Reply:
Fair enough. Let's just not delete them to begin with as a solution?